bot/exts/evergreen/wikipedia.py

```diff
@@ -0,0 +1,111 @@
+            data = await response.json()
+        page = []
+
+        search_results = data["query"]["search"]
+        if not search_results:
+            return None
+
+        # Ignore pages with "may refer to"
+        for search_result in search_results:
+            log.info("trying to append titles")
+            if "may refer to" not in search_result["snippet"]:
+                page.append(search_result["title"])
+        log.info("Finished appending titles")
+        return page
+
+    @commands.cooldown(1, 10, commands.BucketType.user)
+    @commands.command(name="wikipedia", aliases=["wiki"])
+    async def wikipedia_search_command(self, ctx: commands.Context, *, search: str) -> None:
+        """Return list of results containing your search query from wikipedia."""
+        titles = await self.search_wikipedia(search)
+
+        def check(message: Message) -> bool:
+            return message.author.id == ctx.author.id and message.channel == ctx.channel
+
+        if not titles:
+            await ctx.send("Sorry, we could not find a wikipedia article using that search term")
+            return
+
+        async with ctx.typing():
+            titles_no_underscore = [title.replace(" ", "_") for title in titles]  # wikipedia uses "_" as spaces
+            log.info("Finished appending titles to titles_no_underscore list")
+
+            s_desc = "\n".join(self.formatted_wiki_urls(index, title)for index, title in enumerate(titles, start=1))
+            embed = Embed(colour=Color.blue(), title=f"Wikipedia results for `{search}`", description=s_desc)
+            embed.timestamp = datetime.datetime.utcnow()
+            await ctx.send(embed=embed)
+        embed = Embed(colour=Color.green(), description="Enter number to choose")
+        msg = await ctx.send(embed=embed)
+        titles_len = len(titles_no_underscore)  # getting length of list
+
+        for retry_count in range(1, Wikipedia.total_chance + 1):
+            retries_left = Wikipedia.total_chance - retry_count
+            if retry_count < Wikipedia.total_chance:
+                error_msg = f"You have `{retries_left}/{Wikipedia.total_chance}` chances left"
+            else:
+                error_msg = 'Please try again by using `.wiki` command'
+            try:
+                message = await ctx.bot.wait_for('message', timeout=60.0, check=check)
+                response_from_user = await self.bot.get_context(message)
+                if response_from_user.command:
+                    return
+                response = int(message.content)
+                if response < 0:
+                    await ctx.send(f"Sorry, but you can't give negative index, {error_msg}")
+                elif response == 0:
+                    await ctx.send(f"Sorry, please give an integer between `1` to `{titles_len}`, {error_msg}")
+                else:
+                    await ctx.send(WIKIPEDIA_URL.format(title=titles_no_underscore[response - 1]))
```

Comment:
```suggestion
                    await ctx.send(WIKIPEDIA_URL.format(title=titles[response - 1].replace(" ", "_")))
```
This removes the need for the underscores list, which is still named `titles_no_underscore_, which implies the links don't have underscore in it, but it's the opposite.